interp/run.go

```diff
@@ -2036,10 +2034,16 @@ func recv(n *node) {
 	if n.fnext != nil {
 		fnext := getExec(n.fnext)
 		n.exec = func(f *frame) bltn {
-			chosen, v, _ := reflect.Select([]reflect.SelectCase{
-				{Dir: reflect.SelectRecv, Chan: reflect.ValueOf(f.done)},
-				{Dir: reflect.SelectRecv, Chan: value(f)},
-			})
+			ch := value(f)
+			// Fast: channel read doesn't block
+			if x, ok := ch.TryRecv(); ok {
```

Comment:
I don't think this is the right semantics. It changes completely how channel operations work.

Reply:
What is not correct exactly ? I don't see. If `TryRecv` succeeds, no need to `Select` and block, no ?

Reply:
Sorry, you are absolutely correct. I misread the `TryRecv` docs.